include/libcyphal/transport/can/msg_rx_session.hpp

```diff
@@ -25,22 +25,33 @@ namespace can
 ///
 namespace detail
 {
-class MessageRxSession final : public IMessageRxSession, private SessionDelegate
+
+/// @brief A class to represent a message subscriber RX session.
+///
+class MessageRxSession final : public IMessageRxSession, private IRxSessionDelegate
 {
-    // In use to disable public construction.
-    // See https://seanmiddleditch.github.io/enabling-make-unique-with-private-constructors/
-    struct Tag
+    /// @brief Defines specification for making interface unique ptr.
+    ///
+    struct Spec
     {
-        explicit Tag()  = default;
         using Interface = IMessageRxSession;
         using Concrete  = MessageRxSession;
+
+        // In use to disable public construction.
+        // See https://seanmiddleditch.github.io/enabling-make-unique-with-private-constructors/
+        explicit Spec() = default;
     };
 
 public:
     CETL_NODISCARD static Expected<UniquePtr<IMessageRxSession>, AnyError> make(TransportDelegate&     delegate,
                                                                                 const MessageRxParams& params)
     {
-        auto session = libcyphal::detail::makeUniquePtr<Tag>(delegate.memory(), Tag{}, delegate, params);
+        if (params.subject_id > CANARD_SUBJECT_ID_MAX)
+        {
+            return ArgumentError{};
+        }
```

Comment:
Parameters are const (see `params_` below), and their `subject_id` won't change. So, it makes sense fail early (at `make` phase) rather than later during transmissions. The same story for other session factories.